web/client/components/TOC/fragments/settings/VisibilityLimitsForm.jsx

```diff
@@ -0,0 +1,405 @@
+    defaultLimitsType,
+    limitsTypesOptions,
+    dpi,
+    onChange
+}) {
+
+    const [limitsType, setLimitsType] = useState(defaultLimitsType || limitsTypesOptions[0].value);
+
+    const {
+        maxResolution,
+        minResolution,
+        disableResolutionLimits
+    } = layer;
+
+    const dpu = dpi2dpu(dpi, projection);
+
+    function updateOptions(options, newValue) {
+        const values = options.map((option) => option[limitsType]);
+        const isMissing = !isNil(newValue[limitsType]) && values.indexOf(newValue[limitsType]) === -1;
+        return isMissing
+            ? [ newValue, ...options].sort((a, b) => a[limitsType] > b[limitsType] ? -1 : 1)
+            : options;
+    }
+
+    function getResolutionObject(value, type) {
+        if (type === 'scale') {
+            const resolution = value / dpu;
+            return {
+                resolution: resolution,
+                scale: value,
+                zoom: getZoomFromResolution(resolution, resolutions)
+            };
+        }
+        return {
+            resolution: value,
+            scale: value * dpu,
+            zoom: getZoomFromResolution(value, resolutions)
+        };
+    }
+
+    const [options, setOptions] = useState([]);
+    const [maxValue, setMaxValue] = useState(maxResolution && {
+        resolution: maxResolution,
+        scale: maxResolution * dpu,
+        zoom: getZoomFromResolution(maxResolution, resolutions)
+    });
+    const [minValue, setMinValue] = useState(minResolution && {
+        resolution: minResolution,
+        scale: minResolution * dpu,
+        zoom: getZoomFromResolution(minResolution, resolutions)
+    });
+    const [loading, setLoading] = useState(false);
+    const [message, setMessage] = useState();
+    const [error, setError] = useState();
+    const isMounted = useRef();
+
+    useEffect(() => {
+        isMounted.current = true;
+        return () => {
+            isMounted.current = false;
+        };
+    }, []);
+
+    function clearAlters() {
+        setError(undefined);
+        setMessage(undefined);
+    }
+
+    function handleGetCapabilities() {
+        if (layer.type === 'wms' && !loading) {
+            setLoading(true);
+            clearAlters();
+            getLayerCapabilities(layer)
+                .toPromise()
+                .then((capabilities) => {
+                    if (isMounted.current) {
+                        const {
+                            maxScaleDenominator,
+                            minScaleDenominator
+                        } = capabilities;
+                        const newMaxValue = !isNil(maxScaleDenominator)
+                            && getResolutionObject(maxScaleDenominator, 'scale');
+                        const newMinValue = !isNil(minScaleDenominator)
+                            && getResolutionObject(minScaleDenominator, 'scale');
+                        const newMessageId = isNil(maxScaleDenominator) && isNil(minScaleDenominator)
```

Comment:
the message is quite misleading imo

https://user-images.githubusercontent.com/11991428/105369424-d3eac300-5c02-11eb-9dcd-f344fd027d61.mp4

why the fields remains empty if it says "has been updated correctly"?